rclcpp/include/rclcpp/executors.hpp

```diff
@@ -26,6 +29,20 @@ namespace executors
 using rclcpp::executors::multi_threaded_executor::MultiThreadedExecutor;
 using rclcpp::executors::single_threaded_executor::SingleThreadedExecutor;
 
+template<typename FutureT>
+std::shared_future<FutureT> &
+spin_node_until_future_complete(
+  rclcpp::executor::Executor & executor, rclcpp::node::Node::SharedPtr & node_ptr,
+  std::shared_future<FutureT> & future)
+{
+  std::future_status status;
+  do {
+    executor.spin_node_some(node_ptr);
+    status = future.wait_for(std::chrono::seconds(0));
+  } while (status != std::future_status::ready && rclcpp::utilities::ok());
+  return future;
```

Comment:
For method chaining I find it pretty intuitive that the function returns the future. Especially since it looks like that passing a temporary future does not work.

I don't know if others have an opinion on this. We could always defer this to an API review in the future.